src/coreclr/src/jit/layout.cpp

```diff
@@ -410,3 +410,64 @@ ClassLayout* ClassLayout::GetPPPQuirkLayout(CompAllocator alloc)
     return m_pppQuirkLayout;
 }
 #endif // TARGET_AMD64
+
+//------------------------------------------------------------------------
+// AreMatching: check if 2 layouts are the same for copying.
+//
+// Arguments:
+//    layout1 - the first layout;
+//    layout2 - the second layout.
+//
+// Return value:
+//    true if matching, false otherwise.
+//
+// Notes:
+//    Layouts are called matching if they are equal or if
+//    they have the same size and the same GC slots.
+//
+// static
+bool ClassLayout::AreMatching(const ClassLayout* layout1, const ClassLayout* layout2)
+{
+    CORINFO_CLASS_HANDLE clsHnd1 = layout1->GetClassHandle();
+    CORINFO_CLASS_HANDLE clsHnd2 = layout2->GetClassHandle();
+    assert(clsHnd1 != NO_CLASS_HANDLE);
+    assert(clsHnd2 != NO_CLASS_HANDLE);
+
+    if (clsHnd1 == clsHnd2)
+    {
+        return true;
+    }
+
+    if (layout1->GetSize() != layout2->GetSize())
+    {
+        return false;
+    }
+
+    if (layout1->HasGCPtr() != layout2->HasGCPtr())
+    {
+        return false;
+    }
+
+    if (!layout1->HasGCPtr() && !layout2->HasGCPtr())
+    {
+        return false;
```

Comment:
Did you intend for this to be `return true;`?

Reply:
Nice catch, thank you, fixed.